File: string/Solution67.java
class Solution67 {
    public String addBinary(String a, String b) {
    	String result = "";
    	int length = a.length();
    	if(a.length() > b.length()){
    		length = a.length();
    		while(b.length() != a.length()){
    			b = "0" + b;
    		}
    	}
    	if(a.length() < b.length()){
    		length = b.length();
    		while(b.length() != a.length()){
    			a = "0" + a;
    		}
    	}
    	boolean sum = false;
    	for(int i = length - 1; i >= 0; i--){
    		char aa = a.charAt(i);
    		char bb = b.charAt(i);
    		if(aa == '1' && bb == '1'){
    			result = sum ? "1" + result : "0" + result;
    			sum = true;
    		}

            else if(aa == '0' && bb == '0'){
                result = sum ? "1" + result : "0" + result;
                sum = false;
            }

            else{
                if(sum){
                    result = "0" + result;
                    sum = true;
                }
                else{
                    result = "1" + result;
                    sum = false;
                }
            }
    	}
        result = sum ? "1" + result : result;
        return result;
    }
}

File: C++/Solution67.cpp
#include<iostream>
#include<vector>
#include<algorithm>
#include<unordered_map>
#include<unordered_set>
using namespace std;

class Solution67 {
public:
    string addBinary(string a, string b) {
        while(a.length() > b.length()){
            b = '0' + b;
        }
        while(a.length() < b.length()){
            a = '0' + a;
        }
        string res = "";
        int carry = 0;
        for(int i = a.length() - 1; i >= 0; i--){
            int sum = a[i] - '0' + b[i] - '0' + carry;
            if(sum >= 2){
                carry = 1;
                res = ((char) (sum - 2 + '0')) + res;
            }
            else{
                carry = 0;
                res = ((char) (sum + '0')) + res;
            }
        }
        if(carry == 1){
            res = "1" + res;
        }
        return res;
    }
};